src/Samples/Samples.WinForms.MultiplePlayers/Form1.cs

```diff
@@ -0,0 +1,154 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using Vlc.DotNet.Core;
+using Vlc.DotNet.Forms;
+
+namespace Samples.WinForms.MultiplePlayers
+{
+    public partial class Form1 : Form
+    {
+        private readonly List<VlcControl> listOfControls = new List<VlcControl>();
+        private const string StreamParams = ":network-caching=2000";
+        private const string MyVlcLibraryPath = @"C:\Program Files (x86)\VideoLAN\VLC";
+
+        public Form1()
+        {
+            InitializeComponent();
+        }
+
+        private void btnAddPlayer_Click(object sender, EventArgs e)
+        {
+            this.AddPlayer();
+        }
+
+        private void btnRemovePlayer_Click(object sender, EventArgs e)
+        {
+            this.RemovePlayer();
+        }
+
+        private void AddPlayer()
+        {
+            lock (listOfControls)
+            {
+                // Create a player instance.
+                var player = new VlcControl();
+                HookEvents(player);
+                player.VlcLibDirectory = new DirectoryInfo(MyVlcLibraryPath);
+                player.EndInit();
+                player.Size = new Size(200, 200);
+
+                // Add player to the panel.
+                this.panel.Controls.Add(player);
+
+                // Tell the player to play
+                player.SetMedia(this.textBox1.Text, Form1.StreamParams);
+                player.Play();
+                player.Audio.IsMute = true;
+
+                // Add to a list
+                listOfControls.Add(player);
+            }
+        }
+
+        private void RemovePlayer()
+        {
+            lock (listOfControls)
+            {
+                if (!listOfControls.Any()) return;
+
+                var player = listOfControls[0];
+
+                // Remove player from the list
+                listOfControls.Remove(player);
+
+                // Dispose player
+                UnhookEvents(player);
+                player.Dispose();
+            }
+        }
+
+        private void button1_Click(object sender, EventArgs e)
+        {
+            lock (listOfControls)
```

Comment:
I'd replace the whole block with : `listOfControls.FirstOrDefault()?.Stop()`, but I'm ok with that.

Reply:
same as previous.